library/src/main/java/com/novoda/downloadmanager/LiteDownloadMigrationService.java

```diff
@@ -0,0 +1,124 @@
+package com.novoda.downloadmanager;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Binder;
+import android.os.IBinder;
+import android.os.SystemClock;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class LiteDownloadMigrationService extends Service {
+
+    private Migrator v1ToV2Migrator;
+    private ExecutorService executor;
+    private IBinder binder;
+    private Migrator.Callback migrationCallback;
+    private String updateMessage;
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        super.onStartCommand(intent, flags, startId);
+
+        Log.e(getClass().getSimpleName(), "onStartCommand");
+
+        if (v1ToV2Migrator != null) {
+            executor.execute(new Runnable() {
```

Comment:
Moved to `onStartCommand` because this needs to re-execute whenever the service is killed and restarted without binding.